src/environmentbase/networkbase.py

```diff
@@ -347,11 +364,59 @@ def add_vpn_gateway(self,
         else:
             vpn_name = self.__class__.__name__ + 'Gateway'
 
-        gateway = self.template.add_resource(ec2.VPNGateway('vpnGateway',
+        gateway = self.add_resource(ec2.VPNGateway('vpnGateway',
             Type=vpn_conf.get('vpn_type', 'ipsec.1'),
             Tags=[ec2.Tag(key='Name', value=vpn_name)]))
 
-        gateway_connection = self.template.add_resource(ec2.VPCGatewayAttachment('vpnGatewayAttachment',
-            VpcId=self.template.vpc_id,
-            InternetGatewayId=self.template.igw,
+        gateway_connection = self.add_resource(ec2.VPCGatewayAttachment('vpnGatewayAttachment',
+            VpcId=self.vpc_id,
+            InternetGatewayId=self.igw,
             VpnGatewayId=gateway))
+
+
+class NetworkBase(EnvironmentBase):
```

Comment:
Ultimately NetworkBase class should not be necessary.  BaseNetwork class should be moved to a new file inside the /patterns directory and this file 'network.py' should be removed. But these tasks can be done in a follow on commit.